### PR TITLE
Add some more changes to API documentation

### PR DESCRIPTION
**Pull request by liesas**
The documentation now includes the the category filter for the projects endpoint.

Previously the documentation was generated from production, this time it was done locally, so the data is obviously fake

### CODE REVIEW

**Review by fwolfst — COMMENTED** [review]
Mmmh, the example data comes with quite some loss of proper examples (e.g. real world message instead of `null`, more linked datapoints, etc. pp.).
Not yet sure if we really want that. Previously, the dev machines had a anonymized copy of the production data, which generated data closer to real life.
Can we maybe generate on staging? Or enrich the seed data? Lets discuss with CTO and product.

**Comment by averell23** [comment]
Hi @liesas - was there any rationale to use seed (or staging) data instead of production? (Any specific problem that came up?)

I don't think it matters that greatly, but I wasn't aware that something wasn't right with the old method either.

**Comment by liesas** [comment]
@averell23 The seed data was used in order to avoid messing around in production. @riccardo-giomi and I had to make a change to the betterdocs gem (https://github.com/betterplace/betterdocs/pull/9) in order to get the documentation generated correctly

**Comment by averell23** [comment]
@liesas - you had to make a change because some change in production broke the API docs? In general creating the docs should not "mess" anything in production, as the API is read-only.

**Comment by liesas** [comment]
@averell23 Yes, creating the API docs shouldn't mess up anything in production. But because we had to make a change to the gem in order for the docs to be generated correctly, we did not feel comfortable doing that in production. Since it is possible to generate the docs locally we preferred doing that.

As to what is the cause of the docs being generated incorrectly: we looked into it for a little bit and weren't able to find out what had changed. We did find out that all controllers including those that hadn't been touched in a long time were affected, so we concluded it should have to do with the gem. There was a similar problem with it in April (https://github.com/betterplace/betterdocs/pull/8) too. By opening https://github.com/betterplace/betterdocs/pull/9 with the fix that was used here we were hoping to have a discussion about it there

**Comment by averell23** [comment]
@liesas @fwolfst I don't see any change for swapping the generation "source" in this commit, only the result. You should probably avoid to introduce "null" fields, but otherwise there isn't anything inherently problematic to use some seed data. 

Do we have the API generation process documented somewhere?

**Comment by liesas** [comment]
@averell23 It is documented here: https://github.com/betterplace/betterdocs#generating

**Comment by liesas** [comment]
@averell23 and I have discussed that using the seed data is fine

File: sections/fundraising_events_featured_projects_list.md [hunk 1]
```diff
@@ -2,7 +2,7 @@
 # Fundraising Event Featured Projects List
 
 ```Cirru
-GET https://api.betterplace.org/de/api_v4/fundraising_events/19267/featured_projects.json
+GET https://api.betterplace.org/de/api_v4/fundraising_events/1/featured_projects.json
```
Comment (Joerg-Seitz) [inline sections/fundraising_events_featured_projects_list.md:5]: Due to using seeded data locally to generate the docs this now returns 404 record not found. So it probably isn't a good example. Same goes for almost all the other example links. People who consume the API are probably more interested in working examples. Is there a plan to handle this?

File: donation_form/third_party_app_donation_form.md [hunk 2]
```diff
@@ -34,12 +30,9 @@ This is a typical user flow when you use the ThirdPartyApp custom donation form
 4. The donation takes place on betterplace.org
 5. We redirect the user back to your page and provide callback parameters
 
-
 ### Alternative solutions
 
-Before you start considering the ThirdPartyApp custom donation form please make sure to check out the other online donation tools that betterplace.org provides. Especially the integrated donation form (iFrame). For more, please visit the [German](https://www.betterplace.org/c/hilfe/) or [English help area](https://www.betterplace.org/c/help).
-
-
+Before you start considering the ThirdPartyApp custom donation form please make sure to check out the other online donation tools that betterplace.org provides. Especially the integrated donation form (iFrame). For more, please visit the [German](https://support.betterplace.org/hc/de) or [English help area](https://www.betterplace.org/c/help).
```
Comment (Joerg-Seitz) [inline donation_form/third_party_app_donation_form.md:35]: Was the URL change on purpose?

**Pull request closed without merging**